src/targets/gpu/hip_gemm_impl.cpp

```diff
@@ -462,10 +462,9 @@ struct hip_gemm_impl
     int32_t
     validate(context& ctx, const std::vector<argument>& input_args, int32_t solution_idx) // const
     {
-        hipblasStatus_t check_valid(HIPBLAS_STATUS_SUCCESS);
         auto common_args = create_hipblaslt_args_common(ctx, input_args, solution_idx);
-        check_valid      = hipblaslt_invoke(&hipblasLtMatmul, common_args);
-        if(check_valid == HIPBLAS_STATUS_SUCCESS)
+        auto check_valid = hipblaslt_invoke(&hipblasLtMatmul, common_args);
+        if(check_valid != HIPBLAS_STATUS_SUCCESS)
         {
             std::cerr << "WARNING:  tuned solution is invalid; reverting to default" << std::endl;
```

Comment:
How can you ever get here?  hipblaslt_invoke() throws an exception if any result besides HIPBLAS_STATUS_SUCCESS occurs.  Is there code I don't see that catches an exception and takes us back here?

Reply:
I think you are right. Thanks for catching this!

Reply:
I have made the change to `hipblaslt_invoke()` to take an optional parameter that controls if we silently return the status or throw an exception with an error message.